cql3/statements/alter_keyspace_statement.cc

```diff
@@ -50,6 +51,21 @@ void cql3::statements::alter_keyspace_statement::validate(query_processor& qp, c
         if (!bool(_attrs->get_replication_strategy_class()) && !_attrs->get_replication_options().empty()) {
             throw exceptions::configuration_exception("Missing replication strategy class");
         }
+        try {
+            data_dictionary::storage_options current_options = qp.db().find_keyspace(_name).metadata()->get_storage_options();
+            data_dictionary::storage_options new_options = _attrs->get_storage_options();
+            if (new_options.type_string() != current_options.type_string()) {
+                throw exceptions::invalid_request_exception(format("Cannot alter storage options type from {} to {}",
+                        current_options.type_string(), new_options.type_string()));
```

Comment:
Ah, you do check it.

There is risk of someone altering the bucket name and losing access, but I can see valid use cases for it.